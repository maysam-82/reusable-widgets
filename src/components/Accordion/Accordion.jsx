import React, { Fragment, useState } from 'react';

const Accordion = ({ items }) => {
	const [activeIndex, setActiveIndex] = useState(null);
	const onTitleClicked = (id) => {
		setActiveIndex(id);
	};

	const renderItems = items.map(({ id, title, content }) => {
		const activeClass = id === activeIndex ? 'active' : '';
		return (
			<Fragment key={id}>
				<div
					className={`title ${activeClass}`}
					onClick={() => onTitleClicked(id)}
				>
					<i className="dropdown icon"></i>
					{title}
				</div>
				<div className={`content ${activeClass}`}>
					<p>{content}</p>
				</div>
			</Fragment>
		);
	});
	return <div className="ui styled accordion container">{renderItems}</div>;
};

export default Accordion;
