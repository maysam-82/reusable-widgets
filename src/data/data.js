import image1 from '../assets/image/00.png';
import image2 from '../assets/image/01.png';
import image3 from '../assets/image/02.png';
import image4 from '../assets/image/03.png';
import image5 from '../assets/image/04.png';
import image6 from '../assets/image/05.png';
import image7 from '../assets/image/06.png';

export const accordionItems = [
    {
        id: 1,
        title: 'What is React?',
        content: 'React is a front end JavaScript framework.',
    },
    {
        id: 2,
        title: 'What use React?',
        content: 'React is a favorite JavaScript library among enginners.',
    },
    {
        id: 3,
        title: 'How do you use React?',
        content: 'You use React by creating components',
    },
];

export const dropDownOptions = [
    { id: 1, label: 'The Color Red', value: 'red' },
    { id: 2, label: 'The Color Blue', value: 'blue' },
    { id: 3, label: 'The Color Yellow', value: 'yellow' },
    { id: 4, label: 'The Color Green', value: 'green' },
];

export const timelineData = [
    {
        id: 1002,
        label: 'season 2',
        date: '10/11/2020',
        image: image1,
        url: '/',
    },
    {
        id: 1001,
        label: 'season 1',
        date: '10/10/2020',
        image: image2,
        url: '/',
    },
    {
        id: 1005,
        label: 'season 5',
        date: '10/02/2021',
        image: image3,
        url: '/',
    },
    {
        id: 1003,
        label: 'season 3',
        date: '10/12/2020',
        image: image4,
        url: '/',
    },
    {
        id: 1006,
        label: 'season 6',
        date: '10/03/2021',
        image: image5,
        url: '/',
    },
    {
        id: 1004,
        label: 'season 4',
        date: '10/01/2021',
        image: image6,
        url: '/',
    },
    {
        id: 1008,
        label: 'season 8',
        date: '10/04/2022',
        image: image2,
        url: '/',
    },
    {
        id: 1007,
        label: 'season 7',
        date: '10/04/2021',
        image: image7,
        url: '/',
    },
];
