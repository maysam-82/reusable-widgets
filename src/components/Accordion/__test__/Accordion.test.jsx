import React from 'react';
import { shallow } from 'enzyme';
import Accordion from '../Accordion';
import { accordionItems } from '../../../data/data';
describe('Accordion component', () => {
	const component = shallow(<Accordion items={accordionItems} />);
	it('should render Accordion component', () => {
		expect(component).toMatchSnapshot();
	});
	describe('When user Click on title of items', () => {
		beforeEach(() => {
			component.find('.title').at(0).simulate('click');
		});
		it('should add class name `active`', () => {
			expect(component.find('.title').at(0).hasClass('active')).toBeTruthy();
		});
	});
});
