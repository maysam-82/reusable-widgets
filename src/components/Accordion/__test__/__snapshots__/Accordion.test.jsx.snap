// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component should render Accordion component 1`] = `
<div
  className="ui styled accordion container"
>
  <div
    className="title "
    onClick={[Function]}
  >
    <i
      className="dropdown icon"
    />
    What is React?
  </div>
  <div
    className="content "
  >
    <p>
      React is a front end JavaScript framework.
    </p>
  </div>
  <div
    className="title "
    onClick={[Function]}
  >
    <i
      className="dropdown icon"
    />
    What use React?
  </div>
  <div
    className="content "
  >
    <p>
      React is a favorite JavaScript library among enginners.
    </p>
  </div>
  <div
    className="title "
    onClick={[Function]}
  >
    <i
      className="dropdown icon"
    />
    How do you use React?
  </div>
  <div
    className="content "
  >
    <p>
      You use React by creating components
    </p>
  </div>
</div>
`;
